dataset_interfaces/interface.py

```diff
@@ -187,12 +189,51 @@ def load(cls, file_path: Path | str) -> "TestExample":
         d["waits"] = [WaitCreator.unserialise(x) for x in d["waits"]]
         d["example_id"] = file_path.name.removesuffix(".def.json")
         d["dataset_name"] = file_path.parent.name
-        assert d["dataset_name"] in DATASETS_BY_NAME, f"Couldn't find a generator for dataset {d['dataset_name']}."
-        dataset_generator = DATASETS_BY_NAME[d["dataset_name"]]()
         generator_attrs = {"dataset_name", "description", "reset_message"}
         d = {k: d[k] for k in d.keys() if k not in generator_attrs}
         return dataset_generator.create_example(**d)
 
+    def default_waits(self):
+
+        if isinstance(self, DynamicExample):
+            return
+
+        assert len(self.is_question) >= 1, "There are no questions for this test"
+        assert len(self.waits) == 0 or len(self.waits) == len(self.is_question), "Current waits should be empty or the same length as the script"
+        memory_span = self.dataset_generator.memory_span
+
+        # TODO: All this token counting is based on OpenAIs methods, which is not Anthropics - but Aanthopic has no reliable tokeniser released so....
+        script_tokens = get_tokens_for_script(self.script)
+        assert script_tokens < memory_span, f"The script for test {self.dataset_name} is too long (estimated {script_tokens} tokens) for the specified memory span of {memory_span} tokens."
+
+        span_minus_script = memory_span - script_tokens
+
+        # Figure out what our spacing is going to be. Just distribute all the needles and questions across 90% of the memory span
+        token_wait = math.floor(span_minus_script * 0.9) // len(self.is_question)
+
+        # Create the empty waits if there are any.
+        if len(self.waits) == 0:
+            for _ in range(len(self.is_question)):
+                self.waits.append({})
+
+        total_wait_tokens = 0
+
+        for idx, f in enumerate(self.waits[:-1]):
+            if f != {}:
+                continue
+
+            self.waits[idx] = WaitCreator.create_wait(tokens=int(token_wait))
+            total_wait_tokens += token_wait
+
+        assert total_wait_tokens < span_minus_script, "Sum of task waits is higher than the determined memory span."
+        self.waits[-1] = WaitCreator.create_wait()
+
+    def setup_question_ratio(self):
```

Comment:
Is this function used anywhere?

Reply:
Its not! Just a prototype that was left in.